### PR TITLE
Use custom backbone.js

### PR DESCRIPTION
**Pull request by mattpap**
This PR removes all unneeded code from backbone.js, reducing it's line count by over 60% and shaving off 16k from bokeh.min.js. No changes to logic were done at this point. The main point, however, is to reduce cognitive burden.  This PR is an effect of lack of results in fixing issue #4153 (strict events).

### CODE REVIEW

**Comment by bryevdv** [comment]
This is a happy PR

**Comment by bryevdv** [comment]
@mattpap can BB also be removed from [`package.json`](https://github.com/bokeh/bokeh/blob/master/bokehjs/package.json) now as well?

File: bokehjs/package.json [hunk 1]
```diff
@@ -86,7 +86,6 @@
   },
   "dependencies": {
     "underscore": "^1.5.2",
-    "backbone": "^1.1.0",
```
Comment (mattpap) [inline bokehjs/package.json]: @bryevdv, it was removed here.

Comment (bryevdv) [inline bokehjs/package.json]: sorry for the noise I tried searching this page for the wrong thing

**Pull request merged**